leases/schema.py

```diff
@@ -0,0 +1,56 @@
+import graphene
+from django.utils.translation import ugettext_lazy as _
+from graphene_django import DjangoConnectionField, DjangoObjectType
+from graphql_jwt.decorators import login_required, superuser_required
+
+from berth_reservations.exceptions import VenepaikkaGraphQLError
+from resources.schema import BerthNode
+
+from .enums import LeaseStatus
+from .models import BerthLease
+
+LeaseStatusEnum = graphene.Enum.from_enum(LeaseStatus)
+
+
+class BerthLeaseNode(DjangoObjectType):
+    berth = graphene.Field(BerthNode)
+    status = LeaseStatusEnum()
+
+    class Meta:
+        model = BerthLease
+        interfaces = (graphene.relay.Node,)
+
+    @classmethod
+    @login_required
+    def get_node(cls, info, id):
+        node = super().get_node(info, id)
+        if not node:
+            return None
+
+        user = info.context.user
+        # TODO: implement proper permissions
+        if (
+            node.customer and node.customer.user and node.customer.user == user
```

Comment:
can we skip `and node.customer.user`? I think it would work fine with just `node.customer and node.customer.user == user`, since if `node.customer.user` is `None` it would simply return `False` 🤔

Reply:
Good point. You can see the trauma JS left on me 😂 😅

Reply:
ahah, yeah, this smells of JS logic 😛